Vagrantfile

```diff
@@ -126,6 +126,12 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     # Start an OpenShift cluster
     # Currently this only works with the (default) VirtualBox provider.
 
+    # Tag configuration as stale when provisioning a dev cluster to
+    # ensure that nodes can wait for fresh configuration to be generated.
+    if ARGV[0] =~ /^up|provision$/i and not ARGV.include?("--no-provision")
+      system('test -d ./openshift.local.config && touch ./openshift.local.config/.stale')
```

Comment:
A vagrant dev cluster launches the master, which has the origin repo mounted as /data.  The openshift-master unit file specifies /data as the working directory, and when the master starts, it creates configuration for both master and nodes in /data/openshift.local.config.  Locating the configuration on a shared directory allows the node provisioning access to the configuration.

This line, then, is checking whether a dev cluster was previously launched and its configuration is still present.